File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/storage/cache/NodeCache.java
package de.cubbossa.pathfinder.storage.cache;

import de.cubbossa.pathfinder.group.NodeGroup;
import de.cubbossa.pathfinder.node.Node;
import java.util.Collection;
import java.util.Optional;
import java.util.UUID;

public interface NodeCache extends StorageCache<Node> {

  <N extends Node> Optional<N> getNode(UUID uuid);

  Optional<Collection<Node>> getAllNodes();

  CacheCollection<UUID, Node> getNodes(Collection<UUID> ids);

  void writeAll(Collection<Node> nodes);

  void write(NodeGroup group);

  void invalidate(UUID uuid);
}

File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/storage/cache/NodeCache.kt
package de.cubbossa.pathfinder.storage.cache

import de.cubbossa.pathfinder.group.NodeGroup
import de.cubbossa.pathfinder.node.Node
import de.cubbossa.pathfinder.storage.cache.StorageCache.CacheCollection
import java.util.*

interface NodeCache : StorageCache<Node> {
    fun <N : Node> getNode(uuid: UUID): N?

    val allNodes: Collection<Node>?

    fun getNodes(ids: Collection<UUID>): CacheCollection<UUID, Node>

    fun writeAll(nodes: Collection<Node>)

    fun write(group: NodeGroup)

    fun invalidate(uuid: UUID)
}
